pkg/controller/validate/validate.go

```diff
@@ -0,0 +1,29 @@
+package validate
+
+import (
+	"errors"
+
+	"github.com/hashicorp/go-multierror"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+)
+
+func ClusterSpec(clusterSpec mdbv1.AtlasClusterSpec) error {
+	var err error
+	if clusterSpec.AdvancedClusterSpec == nil && clusterSpec.ClusterSpec == nil {
+		err = multierror.Append(err, errors.New("expected exactly one of spec.clusterSpec or spec.advancedClusterSpec, neither were present"))
```

Comment:
Any advantages of using multi-error here?

Reply:
in this specific use case no, since we have only one error ever returned, but in general I think using multi error to include a list of all validation errors is better than returning the first one we encounter.